spec/02-integration/04-admin_api/08-targets_routes_spec.lua

```diff
@@ -943,19 +943,26 @@ describe("Admin API #" .. strategy, function()
       end)
 
       it("is allowed and works", function()
-        ngx.sleep(1)
-        local res = client:patch("/upstreams/" .. upstream.name .. "/targets/" .. target.target, {
-          body = {
-            weight = 659,
-          },
-          headers = { ["Content-Type"] = "application/json" }
-        })
-        assert.response(res).has.status(200)
+        local admin_client = assert(helpers.admin_client())
+        local res
+        assert
+          .with_timeout(10)
+          .eventually(function()
+            res = admin_client:patch("/upstreams/" .. upstream.name .. "/targets/" .. target.target, {
+              body = {
+                weight = 659,
+              },
+              headers = { ["Content-Type"] = "application/json" }
+            })
+            return res and res.status == 200
```

Comment:
The response body is not being consumed in this function, which can lead to weird problems when the function is retried with the same http connection due to some oddities with lua-resty-http:

```lua
-- try #1
res = admin_client:patch(...)

-- res.status was not 200, so let's try again
-- the response body from try #1 is still in our recv buffer

-- try #2
-- lua-resty-http will potentially throw an error here (i.e. 
-- "failed to parse response status") because when it calls 
-- `socket:receive()` it is actually reading the response
-- body from try #1 
res = admin_client:patch(...)
```

You can avoid this condition by explicitly reading the response body (`res:read_body()`) or using an assertion that reads the response body as a side effect (such as `assert.response(res).has.status(200)`) and then change the `is_truthy()` condition to `has_no_error()`.

Reply:
Good catch, thanks. Changed it!